src/System.Runtime.WindowsRuntime/src/System/Resources/WindowsRuntimeResourceManager.cs

```diff
@@ -148,9 +148,10 @@ private static CultureInfo GetBestFitCultureFromLanguageList(List<string> langua
                     return new CultureInfo(languages[i]);
                 }
 
-                if (Interop.Kernel32.ResolveLocaleName(languages[i], localeNameBuffer, localeNameBuffer.MaxCapacity) != 0)
+                int result = Interop.Kernel32.ResolveLocaleName(languages[i], localeNameBuffer, Interop.Kernel32.LOCALE_NAME_MAX_LENGTH); 
+                if (result != 0)
                 {
-                    string localeName = localeNameBuffer.ToString();
+                    string localeName = new string(localeNameBuffer, 0, result - 1); // result length includes null terminator
```

Comment:
>new string(localeNameBuffer, 0, result - 1);  [](start = 40, length = 45)

does String constructor work if we pass 0 for the length? I am asking for the case if we get invariant name which is empty string

Reply:
Yes, if you pass in 0 for the length, you get back String.Empty.